CONTRIBUTING.md

```diff
@@ -144,104 +98,56 @@ into a single one that captures everything you're trying to do -- the history
 will never make for pleasant bedtime reading, so focus instead on making your
 commits useful for tools like `git-blame` and `git-bisect`.
```

Comment:
Potentially worth adding here that it's highly desirable for every individual commit to at least be in a compiling & run-able state. Obviously-broken "wip" commits should be discouraged.